### PR TITLE
this is to enforce the required fields for each module

### PR DESCRIPTION
**Pull request by djdanielsson**
### What does this PR do?
sets "mandatory" for the required fields

### How should this be tested?
I think automated tests should work fine for this.

### Is there a relevant Issue open for this?
no

### Other Relevant info, PRs, etc.
none

### CODE REVIEW

**Comment by ericzolf** [comment]
I never had to use `mandatory`, for me it's only necessary if you want to make a variable mandatory which isn't mandatory according to the module, which isn't the case here. The module would choke for us if the variable isn't defined.
It doesn't really hurt but it's more code to maintain. What would be the added value?

**Comment by djdanielsson** [comment]
clarity when looking at the role which parts are required plus it would also remind the user that its a required field when it errors on them.

File: roles/inventory_sources/tasks/main.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   awx.awx.tower_inventory_source:
-    name:                                 "{{ source.name }}"  # REQUIRED
+    name:                                 "{{ source.name | mandatory }}"
     new_name:                             "{{ source.new_name | default(omit) }}"
     description:                          "{{ source.description | default(omit) }}"
-    inventory:                            "{{ source.inventory.name | default( source.inventory ) }}" # REQUIRED
+    inventory:                            "{{ source.inventory.name | default( source.inventory ) | mandatory }}"
     organization:                         "{{ source.inventory.organization.name | default( source.organization | default(omit)) }}"
-    source:                               "{{ source.source }}"  # REQUIRED
+    source:                               "{{ source.source }}"  # REQUIRED ? according to defaults this is not required
```
Comment (sean-m-sullivan) [inline roles/inventory_sources/tasks/main.yml:9]: After looking at the gui, and doing a test of the API, This is a required field, updating the module to fix it over at awx, good catch!

**Review by sean-m-sullivan — APPROVED** [review]
Looks good!, I've reviewed the changes and all of these are required fields in the modules, so have no issue making them mandatory.

File: roles/inventory_sources/tasks/main.yml (hunk 1 above)
Comment (sean-m-sullivan) [reply roles/inventory_sources/tasks/main.yml:9]: PR here: https://github.com/ansible/awx/pull/9284

**Pull request merged**